### ISSUE TITLE
ftdetect with LSP

### ISSUE THREAD

**Issue by hissssst** [jvirtanen/vim-hcl#14]
### Overview

I have `coc.nvim` LSP setup for terraform which is triggered when `filetype=terraform` or `filetype=tf`
But this plugin sets `filetype=hcl` for `*.tf` files

### Workaround

To have this plugin working with both LSP and hcl syntax I have this in my config
```vim
au BufNewFile,BufRead *.tf     set ft=tf
au BufNewFile,BufRead *.tfvars set ft=tf
au Filetype tf set syntax=hcl
```

### Comments

I could've created PR with this fix, but I am not sure that this is the right way to do this

**Comment by jvirtanen** [comment]
Hi @hissssst! Alright, that's an interesting observation. Clearly, for Nomad and Terraform we _do_ have more precise information about the file type than it just being HCL. Your solution above looks good too. I will investigate a bit how other Vim plugins handle this. 👍🏼

**Comment by jvirtanen** [comment]
Hi @hissssst! Do you think #15 would be helpful for this? 🔍

**Comment by jvirtanen** [comment]
This was closed by #15.
